runtime/scripts/optwin.lua

```diff
@@ -0,0 +1,661 @@
+      table.insert(lines, '\t' .. local_to_buffer)
+    elseif info.scope == 'win' then
+      table.insert(lines, '\t' .. local_to_window)
+    end
+
+    local shortname = info.shortname or name
+    if shortname == '' then
+      shortname = name
+    end
+    if info.type == 'boolean' then
+      if vim.o[name] then
+        table.insert(lines, (' \tset %s\tno%s'):format(shortname, shortname))
+      else
+        table.insert(lines, (' \tset no%s\t%s'):format(shortname, shortname))
+      end
+    else
+      local value = vim.o[name] --[[@as string]]
+      value = vim.fn.escape(value, ' \t\\"|')
+      table.insert(lines, (' \tset %s=%s'):format(shortname, value))
+    end
+  end
+end
+
+---@param line string
+---@return 'set'|'help'|'header'|'other'
+local function line_get_type(line)
+  if line:find('^ \tset ') then
+    return 'set'
+  elseif line:find('^%s-[0-9]') then
+    return 'header'
+  elseif line:find('^[\ta-z]') then
+    return 'help'
+  end
+  return 'other'
+end
+
+---@return integer|false
+local function find_alt_window()
+  if not vim.fn.getline(vim.fn.line('.') - 1):match('%(local to') then
+    return 0
+  end
+
+  local thiswin = vim.fn.winnr()
+  local altwin = vim.fn.winnr('#')
+
+  if vim.bo[vim.fn.winbufnr(altwin)].filetype == 'help' then
+    altwin = altwin + 1
+    if altwin == thiswin then
+      altwin = altwin + 1 --[[@as integer]]
+    end
+  end
+
+  if
+    altwin == 0
+    or altwin == thiswin
+    or vim.bo[vim.fn.winbufnr(altwin)].filetype == 'help'
+    or vim.fn.winnr('$') < altwin
+  then
+    vim.notify("Don't know in which window")
+    return false
+  end
+
+  return vim.fn.win_getid(altwin)
+end
+local function update_current_line()
+  local line = vim.api.nvim_get_current_line()
+  if line_get_type(line) ~= 'set' then
+    return
+  end
+  ---@type string
+  local name
+  if line:find('=') then
+    name = line:match('^ \tset (.-)=')
+  else
+    name = line:match('^ \tset ([a-z]*)'):gsub('^no', '') --[[@as string]]
+  end
+
+  local win = find_alt_window()
+  if not win then
+    return
+  end
+  local value = vim._with({
+    win = win,
+  }, function()
+    return vim.o[name]
+  end)
+
+  if vim.api.nvim_get_option_info2(name, {}).type == 'boolean' then
+    if value then
+      vim.api.nvim_set_current_line((' \tset %s\tno%s'):format(name, name))
+    else
+      vim.api.nvim_set_current_line((' \tset no%s\t%s'):format(name, name))
+    end
+  else
+    value = vim.fn.escape(value, ' \t\\"|')
+    vim.api.nvim_set_current_line((' \tset %s=%s'):format(name, value))
+  end
+end
+
+local buf = vim.fn.bufnr('option-window')
+if buf >= 0 then
+  local winids = vim.fn.win_findbuf(buf)
+  if #winids > 0 and vim.fn.win_gotoid(winids[1]) == 1 then
+    return
+  end
+
+  vim.cmd.split()
+  vim.api.nvim_set_current_buf(buf)
+else
+  vim.cmd((vim.env.OPTWIN_CMD or '') .. ' new option-window')
+
+  buf = vim.api.nvim_get_current_buf()
+  vim.bo[buf].bufhidden = 'wipe'
```

Comment:
I am skeptical about this pattern. It can be quite annoying when a buffer goes missing just because it was hidden briefly. Destroying the buffer with it is closed explicitly via `q` (or `<Plug>(nvim-close-win)`) makes sense but otherwise... why?